Tools/clinic/clinic.py

```diff
@@ -870,8 +874,20 @@ class CLanguage(Language):
             {{"{name}", {methoddef_cast}{c_basename}{methoddef_cast_end}, {methoddef_flags}, {c_basename}__doc__}},
     """)
     GETTERDEF_PROTOTYPE_DEFINE: Final[str] = normalize_snippet(r"""
-        #define {getter_name}    \
-            {{"{name}", (getter){c_basename}, NULL, NULL}},
+        #if defined({getset_name}_GETSETDEF)
+        #   undef {getset_name}_GETSETTDEF
+        #   define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, (setter){getset_basename}_set, NULL}},
+        #else
+        #   define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, NULL, NULL}},
+        #endif
+    """)
+    SETTERDEF_PROTOTYPE_DEFINE: Final[str] = normalize_snippet(r"""
+        #if defined({getset_name}_GETSETDEF)
+        #   undef {getset_name}_GETSETDEF
+        #   define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, (setter){getset_basename}_set, NULL}},
+        #else
+        #   define {getset_name}_GETSETTERDEF {{"{name}", NULL, (setter){getset_basename}_set, NULL}},
+        #endif
```

Comment:
Nitpick: I see the other preprocessor templates are using either 2 or 4 space indents. Let's align with the existing style before merging.

```suggestion
        #  undef {getset_name}_GETSETTDEF
        #  define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, (setter){getset_basename}_set, NULL}},
        #else
        #  define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, NULL, NULL}},
        #endif
    """)
    SETTERDEF_PROTOTYPE_DEFINE: Final[str] = normalize_snippet("""
        #if defined({getset_name}_GETSETDEF)
        #  undef {getset_name}_GETSETDEF
        #  define {getset_name}_GETSETDEF {{"{name}", (getter){getset_basename}_get, (setter){getset_basename}_set, NULL}},
        #else
        #  define {getset_name}_GETSETTERDEF {{"{name}", NULL, (setter){getset_basename}_set, NULL}},
```